internal/pkg/certificates/certificates.go

```diff
@@ -0,0 +1,148 @@
+		if err != nil {
+			return errors.Join(errors.New("unable to PEM encode CA certificate"), err)
+		}
+		err = updateWebhooksCaBundle(ctx, k8s, certificatePEMEncoded.Bytes())
+		if err != nil {
+			return err
+		}
+		logger.Info("Certificates and webhooks updated")
+		logger.Info("Restarting controller due a root certificate authority renewal")
+		os.Exit(constants.ControllerReturnCodeCAInitialized)
+		return nil
+	} else if err != nil {
+		err = errors.Join(fmt.Errorf("cannot get root CA secret: %q", constants.RootCASecretName), err)
+	}
+	return err
+}
+
+func createRootCa(ctx context.Context, k8s client.Client, rootCASecret *corev1.Secret, deploymentsNamespace string) (*controllerCrypto.CA, error) {
+	rootCA, err := controllerCrypto.GenerateCA()
+	if err != nil {
+		return nil, errors.Join(errors.New("unable to create root ca"), err)
+	}
+	rootCASecret.Type = corev1.SecretTypeTLS
+	certificatePEMEncoded, err := rootCA.PEMEncodeCertificate()
+	if err != nil {
+		return nil, errors.Join(errors.New("unable to pem encode ca certificate"), err)
+	}
+	privateKeyPEMEncoded, err := rootCA.PEMEncodePrivateKey()
+	if err != nil {
+		return nil, errors.Join(errors.New("unable to pem encode ca private key"), err)
+	}
+	rootCASecret.Data = map[string][]byte{
+		"tls.crt": certificatePEMEncoded.Bytes(),
+		"tls.key": privateKeyPEMEncoded.Bytes(),
+	}
+	rootCASecret.Name = constants.RootCASecretName
+	rootCASecret.Namespace = deploymentsNamespace
+	err = k8s.Create(ctx, rootCASecret)
+	if err != nil {
+		err = errors.Join(errors.New("cannot create root ca secret"), err)
+	}
+	return rootCA, err
+}
+
+func renewControllerCerts(ctx context.Context, k8s client.Client, ca *controllerCrypto.CA, subjectAlternativeNames []string, deploymentsNamespace string) error {
+	certificate, certificateKey, err := controllerCrypto.GenerateCert(ca.CaCertBytes, subjectAlternativeNames, ca.CaPrivateKey)
+
+	if err != nil {
+		return fmt.Errorf("unable to create controller certificates")
+	}
+	controllerCertSecret := corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      constants.ControllerCertsSecretName,
+			Namespace: deploymentsNamespace,
+		},
+		Immutable:  new(bool),
+		Data:       map[string][]byte{"tls.crt": certificate.Bytes(), "tls.key": certificateKey.Bytes()},
+		StringData: map[string]string{},
+		Type:       corev1.SecretTypeTLS,
+	}
+	err = k8s.Create(ctx, &controllerCertSecret)
+	if apierrors.IsAlreadyExists(err) {
+		err = k8s.Update(ctx, &controllerCertSecret)
+	}
+	if err != nil {
+		return errors.Join(errors.New("unable to create controller certificate secret"), err)
+	}
+	logger.Info("controller certificate secret created")
+	return nil
+}
+
+func updateWebhooksCaBundle(ctx context.Context, k8s client.Client, caBundle []byte) error {
+	webhookConfigList := admissionregistrationv1.ValidatingWebhookConfigurationList{}
+	err := k8s.List(ctx, &webhookConfigList, client.MatchingLabels{"app.kubernetes.io/part-of": "kubewarden"})
```

Comment:
The controller's webhook certs have a hard coded name, we should just do a `Get`